### PR TITLE
[crunchyroll] Fix video_title (closes #25047)

### LINKED ISSUE #25047: [Crunchyroll] Error: Unable to extract video_title
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.03.24**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
/usr/bin/youtube-dl --write-sub --sub-lang deDE --netrc --verbose -o '%(series)s - E%(episode_number)02d - %(episode)s.%(ext)s' -f bestvideo+bestaudio/best -- https://www.crunchyrol
l.com/de/that-time-i-got-reincarnated-as-a-slime/episode-24-black-and-the-mask-780741
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--write-sub', '--sub-lang', 'deDE', '--netrc', '--verbose', '-o', '%(series)s - E%(episode_number)02d - %(episode)s.%(ext)s', '-f', 'bestvideo+bestaudio/best', '--', 'https://www.crunchyroll.com/de/that-time-i-got-reincarnated-as-a-slime/episode-24-black-and-the-mask-780741']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.03.24
[debug] Python version 3.8.2 (CPython) - Linux-4.15.0-96-generic-x86_64-with
[debug] exe versions: ffmpeg 4.2.1, ffprobe 4.2.1
[debug] Proxy map: {}
[crunchyroll] Downloading login page
[crunchyroll] Logging in
[crunchyroll] 780741: Downloading webpage
ERROR: Unable to extract video_title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3.8/site-packages/youtube_dl/extractor/crunchyroll.py", line 444, in _real_extract
    video_title = self._html_search_regex(
  File "/usr/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 1014, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "/usr/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 1005, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract video_title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

## Description

Crunchyroll has updated their website, breaking the video_title regex.
Changing the regex on line 445 of the crunchyroll extractor should fix this:

```
r'(?s)<h1[^>]*>((?:(?!<h1).)*?<span[^>]+itemprop=["\']title["\'][^>]*>(?:(?!<h1).)+?)</h1>'
```
to
```
r'(?s)<h1[^>]*>((?:(?!<h1).)*?<span[^>]+itemprop=[\"\']name[\"\'][^>]*>(?:(?!<h1).)+?)</h1>'
```

### PR DESCRIPTION
**Pull request by LameLemon**
### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Bug fix 
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

The title has been changed so I made the following adjustments to the code:
- Get `video_title` from meta tag instead, avoids inflexible regex
- Add fallback to get `video_title` from `<title>` this is more flexible than the previous regex
- Changed title format in tests to use `, ` instead of  ` – ` as this appears multiple time in the meta and title tags
- Removed test 0 and 1 as the media is no longer available
- Added new test
- Changed one of the test uploader to `Re:Zero Partners` from `TV TOKYO` as this is changed on the site as well

Passes all the single Crunchyroll tests and looks good on flake8. If there is anything else that needs to be done please let me know.

Closes #25047

### CODE REVIEW

File: youtube_dl/extractor/crunchyroll.py [hunk 1]
```diff
@@ -440,11 +423,9 @@ def _real_extract(self, url):
...
             r'(?:vilos\.config\.player\.language|LOCALE)\s*=\s*(["\'])(?P<lang>(?:(?!\1).)+)\1',
             webpage, 'language', default=None, group='lang')
-
-        video_title = self._html_search_regex(
-            r'(?s)<h1[^>]*>((?:(?!<h1).)*?<span[^>]+itemprop=["\']title["\'][^>]*>(?:(?!<h1).)+?)</h1>',
-            webpage, 'video_title')
-        video_title = re.sub(r' {2,}', ' ', video_title)
+        video_title = self._html_search_meta('title', webpage) or self._html_search_regex(r'<title>(?P<title>.+?), - Watch on Crunchyroll</title>',
+                                                                                          webpage, 'video_title', group='title')
+        video_title = re.sub(r', - Watch on Crunchyroll', '', video_title)
```
Comment (p-f) [inline youtube_dl/extractor/crunchyroll.py:428]: This produces invalid results when being called in a non-english speaking country.
At least for me, in germany, titles will have `, - Anschauen auf Crunchyroll` appended (which is the same phrase being cut off here, just in german)
This was not the case before

Comment (LameLemon) [reply youtube_dl/extractor/crunchyroll.py]: Thank you for pointing this out, I have made some changes to the regex. I have also added another test for a pt-PT link.

**Comment by LameLemon** [comment]
I have made changes to the regex so it works on other locales. I have also added a test in pt-PT.

If you would like me to squash the commits, do let me know.

**Comment by Merzav** [comment]
```
 C:\Users\Chidr>"C:\Users\Chidr\AppData\Local\Programs\Python\Python36\Scripts\youtube-dl.exe" --cookies E:\Download\cookies.txt --verbose -f "[height <=? 360][tbr>500]" --sub-lang ruRU --write-sub --sub-format "ass" "https://www.crunchyroll.com/a-certain-scientific-railgun/episode-13-system-those-who-arrive-at-heavens-will-in-an-only-human-body-792619"
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--cookies', 'E:\\Download\\cookies.txt', '--verbose', '-f', '[height <=? 360][tbr>500]', '--sub-lang', 'ruRU', '--write-sub', '--sub-format', 'ass', 'https://www.crunchyroll.com/a-certain-scientific-railgun/episode-13-system-those-who-arrive-at-heavens-will-in-an-only-human-body-792619']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2020.03.24
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.18362
[debug] exe versions: none
[debug] Proxy map: {}
[crunchyroll] 792619: Downloading webpage
Traceback (most recent call last):
  File "__main__.py", line 19, in <module>
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\__init__.py", line 474, in main
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\__init__.py", line 464, in _real_main
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\YoutubeDL.py", line 2019, in download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\YoutubeDL.py", line 797, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\common.py", line 530, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\crunchyroll.py", line 419, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\crunchyroll.py", line 270, in _download_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\common.py", line 794, in _download_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\common.py", line 660, in _download_webpage_handle
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\common.py", line 627, in _request_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\YoutubeDL.py", line 2238, in urlopen
  File "C:\Python\Python34\lib\urllib\request.py", line 464, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 482, in _open
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\utils.py", line 2572, in http_open
  File "C:\Python\Python34\lib\urllib\request.py", line 1183, in do_open
  File "C:\Python\Python34\lib\http\client.py", line 1137, in request
  File "C:\Python\Python34\lib\http\client.py", line 1177, in _send_request
  File "C:\Python\Python34\lib\http\client.py", line 1109, in putheader
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 2276-2280: ordinal not in range(256)
```

```
C:\Users\Chidr>"C:\Users\Chidr\AppData\Local\Programs\Python\Python36\Scripts\youtube-dl.exe" -u XXX -p XXX --verbose -f "[height <=? 360][tbr>500]" --sub-lang ruRU --write-sub --sub-format "ass" "https://www.crunchyroll.com/a-certain-scientific-railgun/episode-13-system-those-who-arrive-at-heavens-will-in-an-only-human-body-792619"
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-u', 'PRIVATE', '-p', 'PRIVATE', '--verbose', '-f', '[height <=? 360][tbr>500]', '--sub-lang', 'ruRU', '--write-sub', '--sub-format', 'ass', 'https://www.crunchyroll.com/a-certain-scientific-railgun/episode-13-system-those-who-arrive-at-heavens-will-in-an-only-human-body-792619']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2020.03.24
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.18362
[debug] exe versions: none
[debug] Proxy map: {}
[crunchyroll] Downloading login page
[crunchyroll] Logging in
[crunchyroll] 792619: Downloading webpage
ERROR: Unable to extract video_title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\YoutubeDL.py", line 797, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\common.py", line 530, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\crunchyroll.py", line 446, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\common.py", line 1014, in _html_search_regex
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpjwbwqymm\build\youtube_dl\extractor\common.py", line 1005, in _search_regex
youtube_dl.utils.RegexNotFoundError: Unable to extract video_title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

Where is the mistake?

**Comment by LameLemon** [comment]
@Merzav Hello, I just gave both of these methods a go and they both work on my machine. Are you sure you are using the fix on my branch that I've submitted for a PR?

**Comment by skid9000** [comment]
I built your pr on my machine it works perfectly here.

**Comment by Merzav** [comment]
Yes that was my fault, everything is working.

**Pull request closed without merging**